scripts/weatherfx.js

```diff
@@ -309,18 +291,28 @@ async function jsonItem() {
 }
 
 // remove all the current fx on the canvas, also stops all the sounds effects that matches the flag weatherfx.audio
-function clearEffects() {
-    canvas.scene.setFlag("weatherfx", "active", false);
-    let src = canvas.scene.getFlag("weatherfx", "audio");
+async function clearEffects() {
+    await canvas.scene.setFlag("weatherfx", "isActive", false);
+    let effectCondition = canvas.scene.getFlag("weatherfx", "effectCondition");
     Hooks.call(particleWeather, []);
     FXMASTER.filters.setFilters([]);
-    for (let [key, sound] of game.audio.playing) {
-        if (sound.src !== src) continue;
-        sound.stop();
+    if (enableSound && effectCondition.hasSound) {
+        let playlist = game.playlists.getName(playlistName);
+        let sound = playlist.sounds.getName(effectCondition.name);
+        if (sound.playing) {
+            playlist.stopSound(sound);
+        }
+
     }
 }
 
-async function weatherfxPlaylist(playlistName) {
+async function weatherfxPlaylistExists() {
+    let playlist = game.playlists?.contents.find((p) => p.name === playlistName);
+    let playlistExists = playlist ? true : false;
```

Comment:
no need for this variable, can just use playlist variable

Reply:
I was worried because maybe there is a small slightest chance of the user already have a playlist with the same name? lol

Reply:
Then you need to make sure that your Playlist name is very unique

Reply:
I am using 'Weather FX Playlist' because it is also easy to recognize for the user, I want the user to be able to change the sounds and volume, etc. I alert though to not change playlist name nor sounds name.

Reply:
Then it's better to set the name for the Playlist in the settings and let the user decide the Playlist he wants to use

Reply:
that's good indeed. The code creates the playlist once the game is ready, so we could throw a pop up with a text box for the user give a new name if the name is already used. (which I think it is very unlikely, but not impossible).
because next thing the code does is adding the songs to the recently created playlist.